### PR TITLE
[#470] Add Institution form for superadmin

### LINKED ISSUE #470: Add Institution form for superadmin
[Preview](https://projects.invisionapp.com/share/S9RJNE8WXUB#/screens/415229204_Ustawienia_-_Instytucje_-_Formularz_Dodawania)

- [x] Select parent institution modal [#471](https://github.com/cyfronet-fid/sat4envi/issues/471)
  - [ ] Search institution
  - [ ] Selected institution as tree

- [ ] Institution form

**IMPORTANT!!! Depend on the ticket [#471](https://github.com/cyfronet-fid/sat4envi/issues/471)**

### LINKED ISSUE #471: Select parent institution modal
[Preview](https://projects.invisionapp.com/share/S9RJNE8WXUB#/screens/416623692)

- [x] Search institution
- [ ] ~~Selected institution as tree~~

### PR DESCRIPTION
**Pull request by danielkryska**
[Preview](https://projects.invisionapp.com/share/S9RJNE8WXUB#/screens/415229204_Ustawienia_-_Instytucje_-_Formularz_Dodawania)

* Add institution form

Closes #470

### CODE REVIEW

**Comment by danielkryska** [comment]
@kmarszalek, @jswk could you please describe to me what API should I use in case of images? 

- Do we use chunking?
- Do we use the URL load of images?
- Is there any documentation of the S3 images load for UI?

**Comment by jswk** [comment]
> @kmarszalek, @jswk could you please describe to me what API should I use in case of images?

Have a look at the SavedView API thumbnails. It's analogous.

File: s4e-web/src/app/views/settings/manage-institutions/parent-institution-modal/parent-institution-modal.component.ts [hunk 1]
```diff
@@ -61,6 +63,12 @@ export class ParentInstitutionModalComponent extends ModalComponent<Institution>
     this.institutionsSearch.setValue('');
   }
 
+  isSelected(institution: Institution) {
+    return !!this.selectedInstitution
+      && this.selectedInstitution.name === institution.name
+      && this.selectedInstitution.slug === institution.slug;
```
Comment (jswk) [inline s4e-web/src/app/views/settings/manage-institutions/parent-institution-modal/parent-institution-modal.component.ts:69]: Wouldn't just slug check suffice?

File: s4e-web/src/app/views/settings/manage-institutions/add-institution/add-institution.component.html [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  <div class="alert alert-danger" *ngIf="form.errors" data-e2e="alert-message-details">
+    <ul>
+      <li *ngFor="let error of (form.errors | errorKeys)" [ngSwitch]="error">
+        <ng-container *ngSwitchDefault i18n>Nieoczekiwany błąd</ng-container>
+      </li>
+    </ul>
+  </div>
+
+  <p><b>Instytucja nadrzędna (rodzic)</b></p>
+  <p>Klikając w poniższe pole można zmienić dla nowo tworzonej instytucji jednostkę narzędną.</p>
```
Comment (jswk) [inline s4e-web/src/app/views/settings/manage-institutions/add-institution/add-institution.component.html:34]: Shouldn't this and other text elements in here have the i18n attribute as well? Some have it, so it seems a little inconsistent to me

File: s4e-web/src/app/views/settings/manage-institutions/parent-institution-modal/parent-institution-modal.component.ts (hunk 1 above)
Comment (danielkryska) [reply s4e-web/src/app/views/settings/manage-institutions/parent-institution-modal/parent-institution-modal.component.ts:69]: Yes it may be

**Pull request merged**